internal/build/helm.go

```diff
@@ -386,20 +386,35 @@ func (h *Helm) composeValues(ctx context.Context, db map[ref]*resource.Resource,
 	return transform.MergeMaps(result, hr.GetValues()), nil
 }
 
-func (h *Helm) getHelmRepositorySecret(ctx context.Context, repository *sourcev1beta2.HelmRepository) (*corev1.Secret, error) {
-	//if repository.Spec.SecretRef == nil {
-	return nil, nil
-	//}
-	/*name := types.NamespacedName{
-		Namespace: repository.GetNamespace(),
+func (h *Helm) getHelmRepositorySecret(ctx context.Context, repository *sourcev1beta2.HelmRepository, db map[ref]*resource.Resource) (*corev1.Secret, error) {
+	if repository.Spec.SecretRef == nil {
+		return nil, nil
+	}
+
+	lookupRef := ref{
+		GroupKind: schema.GroupKind{
+			Group: "",
+			Kind:  "Secret",
+		},
 		Name:      repository.Spec.SecretRef.Name,
+		Namespace: repository.ObjectMeta.Namespace,
 	}
-	var secret corev1.Secret
-	err := r.Client.Get(ctx, name, &secret)
-	if err != nil {
-		return nil, err
+
+	if secret, ok := db[lookupRef]; ok {
+		raw, err := secret.AsYAML()
+		if err != nil {
+			return nil, err
+		}
+
+		obj, _, err := h.opts.Decoder.Decode(raw, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		return obj.(*corev1.Secret), nil
 	}
-	return &secret, nil*/
+
+	return nil, fmt.Errorf("Secret %v/%v not in our db", repository.ObjectMeta.Namespace, repository.Spec.SecretRef.Name)
```

Comment:
```suggestion
	return nil, fmt.Errorf("no repository secret `%v` found for helmrepository `%s/%s`", lookupRef, repository.Namespace, repository.Name)
```

Reply:
manually applied this suggestion, thanks